src/configuration.c

```diff
@@ -812,6 +814,8 @@ struct mptcpd_config *mptcpd_config_create(int argc, char *argv[])
                 && merge_config(config, &def_config)
                 && check_config(config);
 
+        l_queue_destroy((struct l_queue *) sys_config.plugins_to_load,
+                        l_free);
```

Comment:
Technically this should be `free()` not `l_free()` since you allocated the string with `strdup()` instead of `l_strdup()`.  Ultimately, the code ends up being equivalent in this case, but it's not something we should count on.
```suggestion
                        free);
```

Reply:
So everything that is not allocated using ELL should be freed with the normal `free()`?

Reply:
The goal is to avoid potential allocator mismatches.  In this case, the string you allocated through the C library `strdup()` function should to be deallocated with `free()`, as documented in the [strdup(3)](https://man7.org/linux/man-pages/man3/strdup.3.html#DESCRIPTION) man page.

 `l_free()` currently calls `free()` as the underlying implementation, but we should not count on that since it is an implementation detail.  What happens if ELL's allocator changes so that `l_free()` the interface doesn't change but the implementation no longer boils down to a simple call to `free()`?  Can the same behavior as the `free()` supplied by the C library be guaranteed?   `strdup()` should be matched with `free()` to avoid potential allocator mismatches in the future.  Alternatively use `l_strdup()` with `l_free()`.

Reply:
I see.
Thank you for the explication.